### PR TITLE
Change DATE to map to number of days since epoch

### PR DESCRIPTION
**Pull request by cbornet**

### CODE REVIEW

File: pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/TransformFunction.java [hunk 1]
```diff
@@ -319,6 +319,13 @@ private static TransformStep newComputeFieldFunction(ComputeConfig config) {
                 throw new IllegalArgumentException(
                     "Duplicate compute field name detected: " + field.getName());
               }
+              if (field.getType() == ComputeFieldType.DATE
```
Comment (aymkhalil) [inline pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/TransformFunction.java:322]: Is there a need to update the type inference logic to explicitly not infer `DATE` for primitive types?

File: pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/model/ComputeField.java [hunk 2]
```diff
@@ -142,7 +141,6 @@ private Class<?> getJavaType() {
         case BOOLEAN:
           return Boolean.class;
         case DATE:
-          return Date.class;
```
Comment (aymkhalil) [inline pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/model/ComputeField.java:145]: Good thing is primitive types support on compute is not released yet so no one could be using `DATE` + primitive types so we should be safe here.

File: pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/TransformFunction.java [hunk 3]
```diff
@@ -319,6 +319,13 @@ private static TransformStep newComputeFieldFunction(ComputeConfig config) {
...
+                  && ("value".equals(field.getName()) || "key".equals(field.getName()))) {
+                throw new IllegalArgumentException(
```
Comment (aymkhalil) [inline pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/TransformFunction.java:324]: Would the user get this feedback when they create the function via pulsar admin - or just later on in the logs when the function actually initializes?

**Review by aymkhalil — APPROVED** [review]

File: pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/TransformFunction.java (hunk 1 above)
Comment (cbornet) [reply pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/TransformFunction.java:322]: We don't have any function that would infer a DATE atm. And I believe there won't ever be.

File: pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/TransformFunction.java (hunk 3 above)
Comment (cbornet) [reply pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/TransformFunction.java:324]: In the logs.

**Pull request merged**